.github/workflows/ops_pr-post-merge.yaml

```diff
@@ -0,0 +1,14 @@
+# This workflow runs any cleanup or other behavior after a PR has been merged.
+on:
+  pull_request:
+    types:
+      - closed
+
+jobs:
+  cleanup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: clean up source branch
+        uses: jessfraz/branch-cleanup-action@master
```

Comment:
It is interesting that the action has implicit permission to do this delete action.

Reply:
GitHub permissions are rather coarse. If a user has push permissions to a repo, that means they can delete branches.

The exception to this is branches with branch protection enabled. So, this will not delete, say, `development` when promotion PRs are merged.

The default behavior when creating a PR is that you give the repo owners permissions to push to the source branch on your fork. This means that any user with push permission on the target repo will be able to delete the source fork branch unless that branch has branch protections.